src/main.rs

```diff
@@ -23,23 +23,29 @@ mod trace_reader;
 struct Cli {
     /// Path to .json with trace data
     path_to_trace_data: Utf8PathBuf,
+
+    /// Path to the output file
+    #[arg(short, long, default_value = "profile.pb.gz")]
+    output_path: Utf8PathBuf,
 }
 
-fn main() {
+fn main() -> Result<()> {
     let cli = Cli::parse();
 
     let data =
         fs::read_to_string(cli.path_to_trace_data).expect("Failed to read call trace from a file");
     let serialized_trace: CallTrace =
         serde_json::from_str(&data).expect("Failed to deserialize call trace");
-
     let samples = collect_samples_from_trace(&serialized_trace);
     let resources_keys = collect_resources_keys(&samples);
 
     let profile = build_profile(&samples, &resources_keys);
 
-    let path = Path::new("profile.pb.gz");
-    let mut file = fs::File::create(path).unwrap();
+    if let Some(parent) = cli.output_path.parent() {
+        fs::create_dir_all(parent)
```

Comment:
why? i'd just ask the user to create the dir themselves if it doesn't exist, and he wants the specific nesting

Reply:
<img width="472" alt="image" src="https://github.com/software-mansion/cairo-profiler/assets/56825108/5edc37ea-b729-475a-80d5-8d6312bc5e5e">
I prefer being consistent across platforms